m3u8/parser.py

```diff
@@ -280,15 +284,29 @@ def _parse_cueout(line, state):
         state['current_cue_out_duration'] = res.group(1)
         state['current_cue_out_scte35'] = res.group(2)
 
-
-def _parse_cueout_start(line, state, prevline):
+def _cueout_elemental(line, state, prevline):
     param, value = line.split(':', 1)
-    state['current_cue_out_duration'] = value
     res = re.match('.*EXT-OATCLS-SCTE35:(.*)$', prevline)
     if res:
         state['current_cue_out_scte35'] = res.group(1)
+        state['current_cue_out_duration'] = value
+        return True
+    else:
+        return False
 
+def _cueout_envivio(line, state, prevline):
+    param, value = line.split(':', 1)
+    res = re.match('.*DURATION=(.*),.*,CUE="(.*)"', value)
+    if res:
+        state['current_cue_out_duration'] = res.group(1)
+        state['current_cue_out_scte35'] = res.group(2)
+        return True
+    else:
+        return False
 
+def _parse_cueout_start(line, state, prevline):
```

Comment:
Cool, I liked it! I'd suggest (but this is only kinda of a stylish thing, more functional instead of dealing with state inside that function you return the values), be cool to skip this one:

```python
def _cueout_elemental(...):
      param, value = line.split(':', 1)
      res = re.match('.*EXT-OATCLS-SCTE35:(.*)$', prevline)
      if res:
         return (res.group(1), value)
     else:
         return None

# ... the same for the other parser and

_cueout_state = _cueout_elemental(line, state, prevline) or _cueout_envivio(line, state, prevline)
if _cueout_state:
  state['current_cue_out_duration'] = _cueout_state(1)
  state['current_cue_out_scte35'] = _cueout_state(2)
```